scripts/build_locally.py

```diff
@@ -104,6 +105,15 @@ def run(
         # Always builds using oneMKL interfaces for the cuda target
         onemkl_interfaces = True
 
+    if target_hip is not None:
+        if target_hip == "default":
```

Comment:
It is a bit unclear what the use case assumed here? Is it about `python scripts/build_locally.py --target-hip="default"` only?
Then I believe the error message below needs to be rephrase a bit to something like `No default HIP architecture is supported. It must be specified explicitly`.